src/main/java/echo/tasklist/TaskList.java

```diff
@@ -0,0 +1,124 @@
+package echo.tasklist;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+import echo.exception.EchoException;
+import echo.task.Task;
+
+/**
+ * Represents a list of task
+ *
+ * @author Ernest Lim
+ */
+public class TaskList {
+
+    /** ArrayList of the tasks added to TaskList by the user */
+    private ArrayList<Task> taskList;
```

Comment:
Perhaps a more intuitive variable name here? You can consider renaming `taskList` to `tasks` to indicate that it is a collection of `Task` objects.